src/mon/OSDMonitor.cc

```diff
@@ -1128,30 +1129,46 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
 
   // tell me about it
   for (auto i = pending_inc.new_state.begin();
-       i != pending_inc.new_state.end();
-       ++i) {
+    i != pending_inc.new_state.end(); ++i) {
     int s = i->second ? i->second : CEPH_OSD_UP;
+    int32_t osd = i->first;
+    const entity_addr_t cluster_addr = osdmap.get_cluster_addr(osd);
     if (s & CEPH_OSD_UP)
-      dout(2) << " osd." << i->first << " DOWN" << dendl;
+      dout(2) << __func__
+              << "osd." << osd << " "
+              << "cluster addr " << cluster_addr << " DOWN" << dendl;
     if (s & CEPH_OSD_EXISTS)
-      dout(2) << " osd." << i->first << " DNE" << dendl;
-  }
-  for (map<int32_t,entity_addr_t>::iterator i = pending_inc.new_up_client.begin();
-       i != pending_inc.new_up_client.end();
-       ++i) {
-    //FIXME: insert cluster addresses too
-    dout(2) << " osd." << i->first << " UP " << i->second << dendl;
-  }
-  for (map<int32_t,uint32_t>::iterator i = pending_inc.new_weight.begin();
-       i != pending_inc.new_weight.end();
-       ++i) {
-    if (i->second == CEPH_OSD_OUT) {
-      dout(2) << " osd." << i->first << " OUT" << dendl;
-    } else if (i->second == CEPH_OSD_IN) {
-      dout(2) << " osd." << i->first << " IN" << dendl;
-    } else {
-      dout(2) << " osd." << i->first << " WEIGHT " << hex << i->second << dec << dendl;
-    }
+      dout(2) << __func__
+              << "osd." << osd << " "
+              << "cluster addr " << cluster_addr << " DNE" << dendl;
+  }
+  for (std::map<int32_t, entity_addr_t>::iterator i =
+    pending_inc.new_up_client.begin();
+    i != pending_inc.new_up_client.end(); ++i) {
+    int32_t osd = i->first;
+    const entity_addr_t cluster_addr = osdmap.get_cluster_addr(osd);
+    dout(2) << __func__
+            << "osd." << osd << " "
```

Comment:
you want to have a space before `osd` and after `__func__`. I would suggest adding a space directly before `"osd."`, doing the same for the "cluster" line below, and thus you can remove the added space after the `osd` variable (which also reduces the number of string operations on the stream).
Please check the same for other places where you added (or changed) the output, like the chunk below starting on line 1159.